src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -1004,6 +1004,22 @@ public void writeIfStatement_simple() {
         String.format("%s%s%s%s", "if (condition) {\n", "int x = 3;\n", "int x = 3;\n", "}\n"));
   }
 
+  @Test
+  public void writeIfStatement_booleanObjectCondition() {
```

Comment:
This test should go in IfStatementTest, could we please move it?

Reply:
Sure, added `IfStatementTest` and moved the booleanObjectCondition there.